org.eclipse.xtext.common.types/src/org/eclipse/xtext/common/types/descriptions/TypeResourceDescription.java

```diff
@@ -0,0 +1,145 @@
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.xtext.common.types.descriptions;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.xtext.common.types.access.impl.URIHelperConstants;
+import org.eclipse.xtext.naming.QualifiedName;
+import org.eclipse.xtext.resource.IEObjectDescription;
+import org.eclipse.xtext.resource.IReferenceDescription;
+import org.eclipse.xtext.resource.impl.AbstractResourceDescription;
+import org.eclipse.xtext.resource.impl.ChangedResourceDescriptionDelta;
+import org.eclipse.xtext.util.Strings;
+import org.eclipse.xtext.xbase.lib.util.ToStringBuilder;
+
+/**
+ * More or less copied from NameBasedResourceDescription and
+ * TypeResourceDescription to avoid dependency on common.types.ui
+ *
+ * @author Jan Koehnlein - Initial contribution and API
+ */
+public class TypeResourceDescription extends AbstractResourceDescription {
+	public static class ChangedDelta extends ChangedResourceDescriptionDelta {
+		public ChangedDelta(QualifiedName javaTypeName) {
+			super(null, new TypeResourceDescription(javaTypeName));
+		}
+	}
+
+	public static class ClassDescription implements IEObjectDescription {
+		private final QualifiedName qualifiedName;
+
+		@Override
+		public EClass getEClass() {
+			return null;
+		}
+
+		@Override
+		public EObject getEObjectOrProxy() {
+			throw new UnsupportedOperationException("TODO: auto-generated method stub");
+		}
+
+		@Override
+		public URI getEObjectURI() {
+			throw new UnsupportedOperationException("TODO: auto-generated method stub");
+		}
+
+		@Override
+		public QualifiedName getName() {
+			return qualifiedName;
+		}
+
+		@Override
+		public String getUserData(String key) {
+			return null;
+		}
+
+		@Override
+		public String[] getUserDataKeys() {
+			return Strings.EMPTY_ARRAY;
+		}
+
+		public ClassDescription(QualifiedName qualifiedName) {
+			super();
+			this.qualifiedName = qualifiedName;
+		}
+
+		@Override
+		public int hashCode() {
+			return 31 * 1 + (qualifiedName == null ? 0 : qualifiedName.hashCode());
+		}
+
+		@Override
+		public boolean equals(Object obj) {
+			if (this == obj) {
+				return true;
+			}
+			if (obj == null) {
+				return false;
+			}
+			if (getClass() != obj.getClass()) {
+				return false;
+			}
+			ClassDescription other = (ClassDescription) obj;
+			if (qualifiedName == null) {
+				if (other.qualifiedName != null) {
+					return false;
+				}
+			} else if (!qualifiedName.equals(other.qualifiedName)) {
+				return false;
+			}
+			return true;
+		}
+
+		@Override
+		public String toString() {
+			ToStringBuilder b = new ToStringBuilder(this);
```

Comment:
I think we can just write this as a String, looks simple enough

Reply:
The original result would be
```
Foo [
  qualifiedName = foo.bar
]
```
If we don't want to change this it is easier to use ToStringBuilder then writing a string in that format.
I'm tended to leave this.